### PR TITLE
Prevent dealiasing when constructing type info for type aliases

### LINKED ISSUE #1858: Wrong response for type at point on values of type aliases
Given the following (already with clues for the tests):
```
object TypeAliases {
  trait State[S, A]
  type IntState[A] = State[Int, A]
  val intInt@intintstate@State: IntState[Int] = ???
  type IntList = List[Int]
  val int@intlist@List: IntList = ???
}
```

The returned name for `intintstate` would be `State[Int]` and `List` for `intlist`. This happens because `tpe.typeSymbol` is used for constructing the name, which normalizes the typeSymbol and dealiases it. 

This has been annoying me, so PR with a fix coming up.

### PR DESCRIPTION
**Pull request by iravid**
See commit message for details.

Resolves #1858.

### CODE REVIEW

**Comment by iravid** [comment]
I toyed around with excluding anything under `scala.Predef` from this change. It's definitely possible. Wondering if it's more correct.

**Comment by fommil** [comment]
If we perhaps have a rule that we dealias anything matching `scala.*` or `.*Predef.*` (don't forget about the user's custom predefs) that would be a pretty good balance?

**Comment by iravid** [comment]
Sounds good to me. I'll make that change.

Perhaps instead of `.*Predef.*` we can fetch the custom predef from the compiler? Does that exist?

**Comment by fommil** [comment]
not really, there is a patch upstream but everybody who does this disables the system one and manually imports their own `Predef` (keeping the naming convention).

**Comment by iravid** [comment]
@fommil what do you think should be done with values of type `type IntMap[A] = Map[Int, A]`? I'm currently going for full dealiasing: display `Map[Int, A]`

**Comment by fommil** [comment]
I'd go for the rule of whatever the type alias's FQN is, rather than what it points at.

If somebody does `type Wibble = String` I'd want to see `Wibble`.

**Comment by iravid** [comment]
That's a great idea. Thanks.

**Comment by iravid** [comment]
Currently still breaking because the dealiasing is dropping some of the type parameters. Will continue tomorrow :-)

**Comment by iravid** [comment]
The AppVeyor failure seems unrelated. @fommil could you restart it?

Other than that, ready for review :-)

**Comment by fommil** [comment]
rebase needed

**Comment by iravid** [comment]
Rebased, thanks!

File: api/src/main/scala/org/ensime/api/common.scala [hunk 1]
```diff
@@ -23,9 +23,10 @@ object DeclaredAs {
   case object Object extends DeclaredAs('object)
   case object Class extends DeclaredAs('class)
   case object Field extends DeclaredAs('field)
+  case object TypeAlias extends DeclaredAs('typeAlias)
```
Comment (fommil) [inline api/src/main/scala/org/ensime/api/common.scala]: this needs a bump in the protocol version. I'm not sure we want to expose type aliases at the api level.

Comment (iravid) [inline api/src/main/scala/org/ensime/api/common.scala]: Aha, you're right. I don't think this is critical either TBH. It just seemed strange that the `DeclaredAs` value would be `Nil`.

What do you say? Shall we drop it?

Comment (fommil) [inline api/src/main/scala/org/ensime/api/common.scala]: yeah, drop it. This all has to do with values retrieved from the graph.

Comment (iravid) [inline api/src/main/scala/org/ensime/api/common.scala]: Done

**Pull request merged**